clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -1033,16 +1035,22 @@ public RequestFuture<Void> commitOffsetsAsync(final Map<TopicPartition, OffsetAn
             lookupCoordinator().addListener(new RequestFutureListener<Void>() {
                 @Override
                 public void onSuccess(Void value) {
-                    pendingAsyncCommits.decrementAndGet();
-                    doCommitOffsetsAsync(offsets, callback);
-                    client.pollNoWakeup();
+                    try {
+                        doCommitOffsetsAsync(offsets, callback);
+                        client.pollNoWakeup();
+                    } finally {
+                        pendingAsyncCommits.decrementAndGet();
+                    }
                 }
 
                 @Override
                 public void onFailure(RuntimeException e) {
-                    pendingAsyncCommits.decrementAndGet();
-                    completedOffsetCommits.add(new OffsetCommitCompletion(callback, offsets,
+                    try {
+                        completedOffsetCommits.add(new OffsetCommitCompletion(callback, offsets,
                             new RetriableCommitFailedException(e)));
+                    } finally {
+                        pendingAsyncCommits.decrementAndGet();
+                    }
```

Comment:
I understand moving the counter decrement to a `finally` block, but are we _expecting_ that the `add()` method or constructor would throw an exception?

Reply:
No, this is just defensive programming